In [17]:
import pandas as pd
import numpy as np
import matplotlib as plt

In [18]:
def formata_valor(valor):
  return f"{valor:,.0f}".replace(',', '.')

In [19]:
path_vendas = "../data/Vendas.xlsx"
path_vendas_dez = "../data/Vendas_Dez.xlsx"
path_gerentes = "../data/Gerentes.xlsx"
df_vendas = pd.read_excel(path_vendas)
df_vendas_dez = pd.read_excel(path_vendas_dez)
df_gerentes = pd.read_excel(path_gerentes)

In [20]:
frames = [df_vendas, df_vendas_dez]
df_vendas = pd.concat(frames)

display(df_vendas)

Código Venda       Data                     ID Loja           Produto  \
0                1 2019-01-01          Iguatemi Esplanada    Sapato Estampa   
1                1 2019-01-01          Iguatemi Esplanada          Camiseta   
2                1 2019-01-01          Iguatemi Esplanada     Sapato Xadrez   
3                2 2019-01-02              Norte Shopping           Relógio   
4                2 2019-01-02              Norte Shopping      Chinelo Liso   
...            ...        ...                         ...               ...   
7084         69996 2019-12-26  Center Shopping Uberlândia    Short Listrado   
7085         69996 2019-12-26  Center Shopping Uberlândia           Mochila   
7086         69996 2019-12-26  Center Shopping Uberlândia  Pulseira Estampa   
7087         69997 2019-12-26           Ribeirão Shopping   Camisa Listrado   
7088         69997 2019-12-26           Ribeirão Shopping       Short Linho   

      Quantidade  Valor Unitário  Valor Final  
0              1             358          358  
1              2             180          360  
2              1             368          368  
3              3             200          600  
4              1              71           71  
...          ...             ...          ...  
7084           2             102          204  
7085           4             270         1080  
7086           1              87           87  
7087           1             108          108  
7088           2             133          266  

[100999 rows x 7 columns]

In [21]:
display(df_vendas.head(3))
display("Linhas / colunas", df_vendas.shape)

Código Venda       Data             ID Loja         Produto  Quantidade  \
0             1 2019-01-01  Iguatemi Esplanada  Sapato Estampa           1   
1             1 2019-01-01  Iguatemi Esplanada        Camiseta           2   
2             1 2019-01-01  Iguatemi Esplanada   Sapato Xadrez           1   

   Valor Unitário  Valor Final  
0             358          358  
1             180          360  
2             368          368

'Linhas / colunas'

(100999, 7)

In [22]:
print('Dados Estatísticos sobre valores')
display(df_vendas[['Valor Unitário', 'Valor Final']].describe().round(2))

Dados Estatísticos sobre valores


Valor Unitário  Valor Final
count       100999.00    100999.00
mean           191.86       414.63
std            145.41       434.09
min             30.00        30.00
25%            100.00       156.00
50%            155.00       274.00
75%            248.00       524.00
max            750.00      3750.00

In [23]:
display(df_vendas['ID Loja'].nunique())
display(df_vendas['Produto'].nunique())

25

120

As vendas contam com 25 lojas e 120 produtos diferentes em venda

In [24]:
total_vendas_por_lojas = df_vendas.groupby('ID Loja')['Valor Final'].sum().apply(formata_valor)
print('TOTAL VENDIDO POR UNIDADE')
display(total_vendas_por_lojas)

TOTAL VENDIDO POR UNIDADE


ID Loja
Bourbon Shopping SP                 1.726.110
Center Shopping Uberlândia          1.668.921
Iguatemi Campinas                   1.762.419
Iguatemi Esplanada                  1.699.681
Norte Shopping                      1.711.968
Novo Shopping Ribeirão Preto        1.678.225
Palladium Shopping Curitiba         1.721.120
Parque Dom Pedro Shopping           1.631.630
Passei das Águas Shopping           1.649.014
Ribeirão Shopping                   1.707.122
Rio Mar Recife                      1.722.766
Rio Mar Shopping Fortaleza          1.698.430
Salvador Shopping                   1.646.739
Shopping Barra                      1.653.103
Shopping Center Interlagos          1.640.486
Shopping Center Leste Aricanduva    1.682.870
Shopping Eldorado                   1.648.111
Shopping Ibirapuera                 1.635.060
Shopping Iguatemi Fortaleza         1.674.824
Shopping Midway Mall                1.590.441
Shopping Morumbi                    1.586.444
Shopping Recife           

In [25]:
total_produtos_vendidos = df_vendas.groupby('Produto')['Quantidade'].sum().apply(formata_valor)
print("TOTAL DE PRODUTOS VENDIDOS")
display(total_produtos_vendidos)

print("VALOR POR PRODUTO")
total_vendas_por_produto = df_vendas.groupby('Produto')['Valor Final'].sum().apply(formata_valor)
display(total_vendas_por_produto)

TOTAL DE PRODUTOS VENDIDOS


Produto
Bermuda             1.815
Bermuda Estampa     1.726
Bermuda Linho       2.024
Bermuda Liso        1.756
Bermuda Listrado    1.799
                    ...  
Tênis Estampa       1.788
Tênis Linho         1.832
Tênis Liso          1.784
Tênis Listrado      1.836
Tênis Xadrez        1.775
Name: Quantidade, Length: 120, dtype: object

VALOR POR PRODUTO


Produto
Bermuda             272.250
Bermuda Estampa     291.694
Bermuda Linho       394.680
Bermuda Liso        275.692
Bermuda Listrado    293.237
                     ...   
Tênis Estampa       457.728
Tênis Linho         538.608
Tênis Liso          474.544
Tênis Listrado      481.032
Tênis Xadrez        459.725
Name: Valor Final, Length: 120, dtype: object